### PR TITLE
Data migration - Adding SKU recommendation

### PR DESCRIPTION
**Pull request by ashutoshsuman99**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Data migration

File: src/datamigration/azext_datamigration/manual/_help.py [hunk 1]
```diff
@@ -24,6 +24,30 @@
...
+        text: |-
+               az datamigration performance-data-collection --connection-string "Data Source=LabServer.database.net;Initial Catalog=master;Integrated Security=False;User Id=User;Password=password" --output-folder "C:\\PerfCollectionOutput" --number-of-interation 5 --perf-query-interval 10 --static-query-interval 60
+      - name: Collect performance data of a given SQL Server using assessment config file.
+        text: |-
+               az datamigration performance-data-collection --config-file-path "C:\\Users\\user\\document\\config.json"
+"""
+
+helps['datamigration get-sku-recommendation'] = """
+    type: command
+    short-summary: Gives SKU recommendations for Azure SQL offerings.
```
Comment (evelyn-ys) [inline src/datamigration/azext_datamigration/manual/_help.py:41]:
```suggestion
    short-summary: Give SKU recommendations for Azure SQL offerings.
```

File: src/datamigration/azext_datamigration/manual/commands.py [hunk 2]
```diff
@@ -19,6 +19,16 @@ def load_command_table(self, _):
...
 
+    with self.command_group(
+        'datamigration performance-data-collection'
+    ) as g:
+        g.custom_command('', 'datamigration_performance_data_collection')
+
+    with self.command_group(
+        'datamigration get-sku-recommendation'
+    ) as g:
+        g.custom_command('', 'datamigration_get_sku_recommendation')
```
Comment (evelyn-ys) [inline src/datamigration/azext_datamigration/manual/commands.py:30]:
```suggestion
    with self.command_group(
        'datamigration'
    ) as g:
        g.custom_command('performance-data-collection', 'datamigration_performance_data_collection')
        g.custom_command('get-sku-recommendation', 'datamigration_get_sku_recommendation')
```

File: src/datamigration/azext_datamigration/manual/custom.py [hunk 3]
```diff
@@ -93,213 +55,121 @@ def datamigration_assessment(connection_string=None,
...
-def validate_config_file_path(path):
-
-    if not os.path.exists(path):
-        raise InvalidArgumentValueError(f'Invalid config file path: {path}. Please provide a valid config file path.')
+def datamigration_performance_data_collection(connection_string=None,
+                                              output_folder=None,
+                                              perf_query_interval=None,
+                                              static_query_interval=None,
+                                              number_of_interation=None,
+                                              config_file_path=None):
```
Comment (evelyn-ys) [inline src/datamigration/azext_datamigration/manual/custom.py:65]: You can set param default vaules here, for example, `static_query_interval=3600`. And the help message will be auto-appended saying "Default: 3600"

**Comment by evelyn-ys** [comment]
Please add tests for new commands

File: src/datamigration/azext_datamigration/manual/custom.py [hunk 4]
```diff
@@ -93,213 +55,121 @@ def datamigration_assessment(connection_string=None,
 
 
 # -----------------------------------------------------------------------------------------------------------------
-# Assessment helper function to test whether the given cofig_file_path is valid and has valid action specified.
```
Comment (sagangwar) [inline src/datamigration/azext_datamigration/manual/custom.py:96]: plz correct the type here: cofig_file_path

File: src/datamigration/azext_datamigration/manual/custom.py (hunk 3 above)
Comment (ashutoshsuman99) [reply src/datamigration/azext_datamigration/manual/custom.py]: Done

**Comment by ashutoshsuman99** [comment]
> Please add tests for new commands

Adding test to these commands would be difficult as currently these cmdlets deal with local computer and not API calls and are only supported in windows environment with some additional setup. 
Please let me know if there is any other way to work with these.

File: src/datamigration/azext_datamigration/manual/commands.py (hunk 2 above)
Comment (ashutoshsuman99) [reply src/datamigration/azext_datamigration/manual/commands.py]: done

**Review by evelyn-ys — APPROVED** [review]

**Comment by evelyn-ys** [comment]
If you want to release a new version, don't forget to update the version in `setup.py` and `history.rst`

**Comment by ashutoshsuman99** [comment]
> If you want to release a new version, don't forget to update the version in `setup.py` and `history.rst`

One question here @evelyn-ys, updating the version will retain the previous version right. Like if the customers wants to install version 0.1.0 they will be able to do so right?

**Comment by evelyn-ys** [comment]
> > If you want to release a new version, don't forget to update the version in `setup.py` and `history.rst`
> 
> One question here @evelyn-ys, updating the version will retain the previous version right. Like if the customers wants to install version 0.1.0 they will be able to do so right?

Exactly. All the versions are available for customers

**Pull request merged**